src/node_i18n.cc

```diff
@@ -62,6 +72,296 @@ bool flag_icu_data_dir = false;
+void Convert(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  CHECK(args[1]->IsString());
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[2]);
+
+  Utf8Value to_name(env->isolate(), args[0]);
+  Utf8Value from_name(env->isolate(), args[1]);
+
+  SPREAD_ARG(args[2], ts_obj);
+
+  UErrorCode status = U_ZERO_ERROR;
+
+  MaybeStackBuffer<char> buf;
+  uint32_t len, limit;
+  char* target;
+  const char* data = ts_obj_data;
+
+  Converter to(env, *to_name);
+  Converter from(env, *from_name);
+  if (to.conv == nullptr || from.conv == nullptr)
+    return;
+
+  ucnv_setSubstChars(to.conv, "?", 1, &status);
+  status = U_ZERO_ERROR;
+
+  limit = ts_obj_length * ucnv_getMaxCharSize(to.conv);
+  buf.AllocateSufficientStorage(ts_obj_length * ucnv_getMaxCharSize(to.conv));
+  target = *buf;
+
+  ucnv_convertEx(to.conv, from.conv,
+                 &target, target + limit,
+                 &data, ts_obj_data + ts_obj_length,
+                 NULL, NULL, NULL, NULL,
+                 true, true,
+                 &status);
+
+  if (U_SUCCESS(status)) {
+    len = target - *buf;
+    args.GetReturnValue().Set(
+      AsBuffer(env->isolate(), &buf, len).ToLocalChecked());
+  } else {
+    ThrowICUError(env, status, "Unable to transcode buffer");
+  }
+}
+
+// Converts to UCS2 from ISO-8859-1 and US-ASCII
+// args[0] is the encoding to convert from
+// args[1] must be a buffer instance
+void ConvertToUcs2(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[1]);
+
+  Utf8Value name(env->isolate(), args[0]);
+
+  SPREAD_ARG(args[1], ts_obj);
+
+  UErrorCode status = U_ZERO_ERROR;
+  size_t len;
+
+  MaybeStackBuffer<UChar> buf(ts_obj_length);
+
+  Converter from(env, *name);
+  if (from.conv == nullptr)
+    return;
+
+  len = ucnv_toUChars(from.conv,
+                      *buf, ts_obj_length << 1,
+                      ts_obj_data, ts_obj_length,
+                      &status);
+
+  if (U_FAILURE(status)) {
+    ThrowICUError(env, status, "Unable to transcode buffer");
+    return;
+  }
+  args.GetReturnValue().Set(
+    AsBuffer(env->isolate(), &buf, ts_obj_length << 1).ToLocalChecked());
+}
+
+// Convert to a named encoding from UCS2
+// args[0] is the name of the encoding being converted to
+// args[1] must be a buffer instance
+void ConvertFromUcs2(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[1]);
+
+  Utf8Value name(env->isolate(), args[0]);
+  SPREAD_ARG(args[1], ts_obj);
+
+  UErrorCode status = U_ZERO_ERROR;
+
+  UChar* source = nullptr;
+  MaybeStackBuffer<UChar> swapspace;
+  const size_t length = ts_obj_length >> 1;
+
+  if (IsLittleEndian()) {
+    source = reinterpret_cast<UChar*>(ts_obj_data);
```

Comment:
Don't do this.  It violates strict aliasing and is going to SIGBUS on architectures that don't support unaligned memory access.